apps/prairielearn/elements/pl-drawing/pl-drawing.py

```diff
@@ -494,3 +496,110 @@ def grade(element_html: str, data: pl.QuestionData) -> None:
         "weight": 1,
         "feedback": {"correct": (score == 1), "missing": {}, "matches": matches},
     }
+
+
+def test(element_html: str, data: pl.ElementTestData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    gradable = pl.get_boolean_attrib(
+        element, "gradable", defaults.element_defaults["gradable"]
+    )
+    if not gradable:
+        return
+
+    # Get raw correct answer
+    name = pl.get_string_attrib(
+        element, "answers-name", defaults.element_defaults["answers-name"]
+    )
+    a_tru = data["correct_answers"][name]
+    grid_size = pl.get_integer_attrib(
+        element, "grid-size", defaults.element_defaults["grid-size"]
+    )
+    tol = pl.get_float_attrib(element, "tol", grid_size / 2)
+    angtol = pl.get_float_attrib(
+        element, "angle-tol", defaults.element_defaults["angle-tol"]
+    )
+
+    result = data["test_type"]
+    if result == "correct" and len(a_tru) > 0:
+        data["raw_submitted_answers"][name] = json.dumps(a_tru)
+        data["partial_scores"][name] = {
+            "score": 1,
+            "weight": 1,
+            "feedback": {
+                "correct": True,
+                "missing": {},
+                "matches": {
+                    element["id"]: True
+                    for element in a_tru
+                    if elements.is_gradable(element["gradingName"])
+                    and element["graded"]
+                },
+            },
+        }
+
+    elif result == "incorrect" and len(a_tru) > 0:
+        data["raw_submitted_answers"][name] = copy.deepcopy(a_tru)
+        for i, element in enumerate(a_tru):
+            if (
+                not elements.is_gradable(element["gradingName"])
+                or not element["graded"]
+            ):
+                continue
+
+            mutated = False
+
+            def get_oob_tol(x: float) -> float:
```

Comment:
"oob" doesn't mean anything to me, is there a better name that can be used here?

Reply:
"out of bounds" - changed in [a3c11f6](https://github.com/PrairieLearn/PrairieLearn/pull/11868/commits/a3c11f697868389e6ae9897ae1af64fe3a7bdccc).